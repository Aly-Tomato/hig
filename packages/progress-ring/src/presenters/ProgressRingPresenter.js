import React from "react";
import PropTypes from "prop-types";

import SVGPresenter from "./SVGPresenter";

import Small from "./smallSVG";
import Medium from "./mediumSVG";
import ExtraSmall from "./xsmallSVG";
import availableSizes from "../availableSizes";

import "./ProgressRingPresenter.scss";

const sizes = {
  xs: {
    svg: ExtraSmall,
    size: 20
  },
  s: {
    svg: Small,
    size: 28
  },
  m: {
    svg: Medium,
    size: 72
  },
  l: {
    svg: Medium,
    size: 144
  },
  xl: {
    svg: Medium,
    size: 242
  }
};

export default function ProgressRingPresenter(props) {
  const { innerRef, percentComplete, size } = props;
  const SVG = sizes[size].svg;
  const originalSize = sizes[size].svg === Medium ? 72 : sizes[size].size;

  return (
    <div
      className="hig__progress-ring"
      role="progressbar"
      aria-valuemin="0"
      aria-valuemax="100"
      aria-valuenow={percentComplete}
      ref={innerRef}
    >
      <SVGPresenter
        height={sizes[size].size}
        width={sizes[size].size}
        original={originalSize}
        svgData={SVG}
      />
    </div>
  );
}

ProgressRingPresenter.defaultProps = { size: "m" };

ProgressRingPresenter.propTypes = {
  innerRef: PropTypes.func,
  percentComplete: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  size: PropTypes.oneOf(availableSizes)
};
