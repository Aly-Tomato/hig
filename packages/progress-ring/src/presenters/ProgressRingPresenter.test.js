import React from "react";
import renderer from "react-test-renderer";
import ProgressRingPresenter from "./ProgressRingPresenter";

describe("progress-ring/presenters/ProgressRingPresenter", () => {
  [
    {
      description: "renders without props",
      props: {}
    },
    {
      description: "renders with all props",
      props: {
        percentComplete: 33,
        size: "s"
      }
    }
  ].forEach(({ description, props: { ...otherProps } }) => {
    it(description, () => {
      const presenter = <ProgressRingPresenter {...otherProps} />;
      const tree = renderer.create(presenter).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
