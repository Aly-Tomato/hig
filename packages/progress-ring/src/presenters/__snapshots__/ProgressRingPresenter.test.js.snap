// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progress-ring/presenters/ProgressRingPresenter renders with all props 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow={33}
  className="hig__progress-ring"
  role="progressbar"
>
  <test-file-stub
    height={28}
    width={28}
  />
</div>
`;

exports[`progress-ring/presenters/ProgressRingPresenter renders without props 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow={undefined}
  className="hig__progress-ring"
  role="progressbar"
>
  <test-file-stub
    height={72}
    width={72}
  />
</div>
`;
